lib/mix/lib/mix/dep.ex

```diff
@@ -167,10 +167,10 @@ defmodule Mix.Dep do
     do: "lock mismatch: the dependency is out of date (run \"mix deps.get\" to fetch locked version)"
```

Comment:
@ericmj may I know what the other the other possibilities could be?

if the dependency is outdated, shouldn't it be caught by: :lockoutdated
so, agreeing with what you say, and that there are other possibilities, why showing an "out of date" message when it may not necessarily be that. I find it misleading. that's what I proposed "the dependency does not match the requirements in mix.lock"